packages/api/cms-api/src/warnings/warning.service.ts

```diff
@@ -34,18 +37,45 @@ export class WarningService {
         );
     }
 
-    public async updateWarningsForBlock(warnings: BlockWarning[], sourceInfo: WarningSourceInfo): Promise<void> {
+    public async saveWarnings({
+        warnings,
+        type,
+        sourceInfo,
+    }: {
+        warnings: CreateWarningInput[];
+        type: string;
+        sourceInfo: WarningSourceInfo;
+    }): Promise<void> {
+        for (const warning of warnings) {
+            await this.saveWarning({
+                warning,
+                type,
+                sourceInfo,
+            });
+        }
+    }
+
+    public async saveWarningsAndDeleteOutdated({
```

Comment:
Do we need this method at all? Why not use the two existing methods instead?

Reply:
I thought it would be easier to make your own CreateWarningService. So you can't forget that old warnings have to be deleted.

Now that I think about it again, I think it would even make sense not to pass the other two functions to the project at all, so that you always have to use this function and can only use the other two in the library.